src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -492,36 +574,23 @@ private void InheritNullableStateOfFieldOrProperty(int targetContainerSlot, int
                     int targetMemberSlot = GetOrCreateSlot(fieldOrProperty, targetContainerSlot);
                     if (targetMemberSlot >= this.State.Capacity) Normalize(ref this.State);
 
+                    bool? value = !fieldOrPropertyType.IsNullable;
                     if (isByRefTarget)
                     {
-                        // This is a property/field acesses through a by ref entity and it isn't considered declared as not-nullable. 
+                        // This is a property/field access through a by ref entity and it isn't considered declared as not-nullable. 
                         // Since reference can point to the heap, we cannot assume the property/field doesn't have null value after this assignment,
                         // regardless of what value is being assigned.
-                        this.State[targetMemberSlot] = (fieldOrPropertyType.IsNullable == true) ? (bool?)false : null;
                     }
                     else if (valueContainerSlot > 0)
                     {
                         int valueMemberSlot = VariableSlot(fieldOrProperty, valueContainerSlot);
-                        this.State[targetMemberSlot] = valueMemberSlot > 0 && valueMemberSlot < this.State.Capacity ?
```

Comment:
>valueMemberSlot < this.State.Capacity [](start = 78, length = 37)

Why is it safe to drop this condition?

Reply:
I can imagine that we have a slot, but the current state hasn't been normalized yet.

---
In reply to: [152698302](https://github.com/dotnet/roslyn/pull/23063#discussion_r152698302) [](ancestors = 152698302)

Reply:
We only add slots in `GetOrCreateSlot()` and that method calls `Normalize()`. We should be able to remove all similar checks of `this.State.Capacity`.

Reply:
> We only add slots in GetOrCreateSlot() and that method calls Normalize(). We should be able to remove all similar checks of this.State.Capacity.

I do not believe this is safe assumptions. It is true only if we didn't switch the state after the slot was added.

Reply:
Reverted.